staging/src/k8s.io/apiserver/pkg/endpoints/handlers/watch.go

```diff
@@ -183,6 +188,10 @@ func (s *WatchServer) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 	defer cleanup()
 	defer s.Watching.Stop()
 
+	t := time.NewTimer(30 * time.Second)
```

Comment:
just as a point of reference, adding a heartbeat watch event to every watcher on this interval means that on a large cluster (5k nodes), this would make the API server write ~700 watch events per second as a baseline even when no objects were changing (node/pod/service/endpoint watchers * 5000 nodes / 30 seconds)

Reply:
Yes, that has to divide by apiservers number. ie. 3 masters cluster, with 700/3 heartbeat/second

Reply:
This can be reduce by only sending heartbeat when idle,  if no other event sent during 30s, server send heartbeat automatically.